src/renderers/shared/reconciler/ReactReconciler.js

```diff
@@ -65,8 +65,10 @@ var ReactReconciler = {
     internalInstance, nextElement, transaction, context
   ) {
     var prevElement = internalInstance._currentElement;
-
-    if (nextElement === prevElement && nextElement._owner != null) {
+    if (nextElement === prevElement &&
+        nextElement._owner != null
+        // TODO: Shouldn't we need to do this: `&& context === internalInstance._context`
```

Comment:
Sure, see https://github.com/facebook/react/compare/pull/4221/head...andreypopp:pr/4221

pr/4221 is a branch on top of this PR and the test case I added fails while I expect it to succeed.